examples/linux-timerfd.rs

```diff
@@ -24,7 +25,12 @@ fn main() -> std::io::Result<()> {
 
         // When the OS timer fires, a 64-bit integer can be read from it.
         Async::new(timer)?
-            .read_with(|t| nix::unistd::read(t.as_raw_fd(), &mut [0u8; 8]).map_err(io::Error::from))
+            .read_with(|t| {
+                // Safety: Assume `as_raw_fd()` returns a valid fd; when `AsFd`
+                // is stabilized, we can remove this unsafe and simplify.
+                let fd = unsafe { BorrowedFd::borrow_raw(t.as_raw_fd()) };
```

Comment:
`AsFd` is stabilized now, so I think we can eliminate this `unsafe`.

Reply:
Almost, but async-io still has an MSRV of 1.48, and I/O safety is stabilized in Rust 1.63. I've now added more to these comments to mention that.

Reply:
We don't run the tests on the MSRV build, so it's alright.